pkg/discover/cloud_foundry/cf_types.go

```diff
@@ -0,0 +1,159 @@
+}
+type AppManifest struct {
+	Name               string                `yaml:"name"`
+	Buildpacks         []string              `yaml:"buildpacks,omitempty"`
+	Docker             *AppManifestDocker    `yaml:"docker,omitempty"`
+	Env                map[string]string     `yaml:"env,omitempty"`
+	RandomRoute        bool                  `yaml:"random-route,omitempty"`
+	NoRoute            bool                  `yaml:"no-route,omitempty"`
+	Routes             *AppManifestRoutes    `yaml:"routes,omitempty"`
+	Services           *AppManifestServices  `yaml:"services,omitempty"`
+	Sidecars           *AppManifestSideCars  `yaml:"sidecars,omitempty"`
+	Processes          *AppManifestProcesses `yaml:"processes,omitempty"`
+	Stack              string                `yaml:"stack,omitempty"`
+	Metadata           *AppMetadata          `yaml:"metadata,omitempty"`
+	AppManifestProcess `yaml:",inline"`
+}
+
+type AppManifestProcesses []AppManifestProcess
+
+type AppManifestProcess struct {
+	Type                             AppProcessType     `yaml:"type,omitempty"`
+	Command                          string             `yaml:"command,omitempty"`
+	DiskQuota                        string             `yaml:"disk_quota,omitempty"`
+	HealthCheckType                  AppHealthCheckType `yaml:"health-check-type,omitempty"`
+	HealthCheckHTTPEndpoint          string             `yaml:"health-check-http-endpoint,omitempty"`
+	HealthCheckInvocationTimeout     uint               `yaml:"health-check-invocation-timeout,omitempty"`
+	Instances                        *uint              `yaml:"instances,omitempty"`
+	LogRateLimitPerSecond            string             `yaml:"log-rate-limit-per-second,omitempty"`
+	Memory                           string             `yaml:"memory,omitempty"`
+	Timeout                          uint               `yaml:"timeout,omitempty"`
+	HealthCheckInterval              uint               `yaml:"health-check-interval,omitempty"`
+	ReadinessHealthCheckType         AppHealthCheckType `yaml:"readiness-health-check-type,omitempty"`
+	ReadinessHealthCheckHttpEndpoint string             `yaml:"readiness-health-check-http-endpoint,omitempty"`
+	ReadinessHealthInvocationTimeout uint               `yaml:"readiness-health-invocation-timeout,omitempty"`
+	ReadinessHealthCheckInterval     uint               `yaml:"readiness-health-check-interval,omitempty"`
+	Lifecycle                        string             `yaml:"lifecycle,omitempty"`
+}
+
+type AppManifestDocker struct {
+	Image    string `yaml:"image,omitempty"`
+	Username string `yaml:"username,omitempty"`
+}
+
+type AppManifestServices []AppManifestService
+
+type AppManifestService struct {
+	Name        string                 `yaml:"name"`
+	BindingName string                 `yaml:"binding_name,omitempty"`
+	Parameters  map[string]interface{} `yaml:"parameters,omitempty"`
+}
+
+func (ams *AppManifestService) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
Quick question here: will this be needed for other CF types?

Reply:
No, this is specific for CF Services due to the multiple ways services can be represented in the CF manifest.